EventFilter/CTPPSRawToDigi/interface/CTPPSPixelRawToDigi.h

```diff
@@ -12,22 +12,22 @@
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
-
 #include "CondFormats/DataRecord/interface/CTPPSPixelDAQMappingRcd.h"
 #include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelDAQMapping.h"
 
 #include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
 #include "FWCore/Framework/interface/ConsumesCollector.h"
 
 
-
 class CTPPSPixelRawToDigi : public edm::stream::EDProducer<> {
 public:
 
   explicit CTPPSPixelRawToDigi( const edm::ParameterSet& );
 
   ~CTPPSPixelRawToDigi() override;
 
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
```

Comment:
Should this be static? If data are filled for all streams there should be a protection against data race, if not, it should not be static.

Reply:
It seems to me that (almost?) all the subdetectors have this declared as static in their unpacking code. I naively used the same approach because I don't see why ctpps pixels should behave differently. If I need to add some protection, could you please point me to some example or be more specific? Thanks

Reply:
yes, it should be static.
This is called by the framework without module instantiation